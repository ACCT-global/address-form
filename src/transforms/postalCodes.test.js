import {
  statePostalCodes,
  citiesPostalCodes,
  neighborhoodPostalCodes,
} from './postalCodes'

describe('Postal Code Transforms', () => {
  it('should transform state postal codes', () => {
    const mapOfStateAndCities = {
      StateName01: {
        City01: '1',
        City02: '1',
      },
      StateName02: {
        City03: '2',
      },
    }

    const result = statePostalCodes(mapOfStateAndCities)

    expect(result).toMatchSnapshot()
  })

  it('should transform cities postal codes', () => {
    const mapOfStateAndCities = {
      StateName01: {
        City01: '1',
        City02: '2',
      },
      StateName02: {
        City03: '3',
      },
    }

    const result = citiesPostalCodes(mapOfStateAndCities)

    expect(result).toMatchSnapshot()
  })

  it('should transform neighborhoods postal codes', () => {
    const mapOfStateAndCities = {
      StateName01: {
        City01: {
          Neighborhood1: '1',
          Neighborhood2: '2',
        },
        City02: {
          Neighborhood3: '3',
          Neighborhood4: '4',
        },
      },
      StateName02: {
        City03: {
          Neighborhood5: '5',
        },
      },
    }

    const result = neighborhoodPostalCodes(mapOfStateAndCities)

    expect(result).toMatchSnapshot()
  })
})
