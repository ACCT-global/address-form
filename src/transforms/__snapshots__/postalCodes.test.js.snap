// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Postal Code Transforms should transform cities postal codes 1`] = `
Object {
  "StateName01": Array [
    Object {
      "city": "City01",
      "postalCode": "1",
    },
    Object {
      "city": "City02",
      "postalCode": "2",
    },
  ],
  "StateName02": Array [
    Object {
      "city": "City03",
      "postalCode": "3",
    },
  ],
}
`;

exports[`Postal Code Transforms should transform neighborhoods postal codes 1`] = `
Object {
  "StateName01": Object {
    "City01": Array [
      Object {
        "neighborhood": "Neighborhood1",
        "postalCode": "1",
      },
      Object {
        "neighborhood": "Neighborhood2",
        "postalCode": "2",
      },
    ],
    "City02": Array [
      Object {
        "neighborhood": "Neighborhood3",
        "postalCode": "3",
      },
      Object {
        "neighborhood": "Neighborhood4",
        "postalCode": "4",
      },
    ],
  },
  "StateName02": Object {
    "City03": Array [
      Object {
        "neighborhood": "Neighborhood5",
        "postalCode": "5",
      },
    ],
  },
}
`;

exports[`Postal Code Transforms should transform state postal codes 1`] = `
Array [
  Object {
    "postalCode": "1",
    "state": "StateName01",
  },
  Object {
    "postalCode": "2",
    "state": "StateName02",
  },
]
`;
